core/src/main/java/brooklyn/management/internal/BasicExternalConfigSupplierRegistry.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package brooklyn.management.internal;
+
+import java.lang.reflect.Constructor;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import brooklyn.config.ConfigMap;
+import brooklyn.config.ConfigPredicates;
+import brooklyn.config.ConfigUtils;
+import brooklyn.config.external.ExternalConfigSupplier;
+import brooklyn.management.ManagementContext;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Simple registry implementation.
+ *
+ * Permits a number of {@link ExternalConfigSupplier} instances to be registered, each with a unique name, for future
+ * (deferred) lookup of configuration values.
+ */
+public class BasicExternalConfigSupplierRegistry implements ExternalConfigSupplierRegistry {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BasicExternalConfigSupplierRegistry.class);
+
+    private final Map<String, ExternalConfigSupplier> providersByName = Maps.newLinkedHashMap();
+
+    public BasicExternalConfigSupplierRegistry(ManagementContext mgmt) {
+        updateFromBrooklynProperties(mgmt);
+    }
+
+    synchronized public void addProvider(String name, ExternalConfigSupplier provider) {
+        if (providersByName.containsKey(name))
+            throw new IllegalArgumentException("Provider already registered with name '" + name + "'");
+        providersByName.put(name, provider);
+    }
+
+    synchronized public void removeProvider(String name) {
+        providersByName.remove(name);
+    }
+
+    /**
+     * Searches the named {@link ExternalConfigSupplier} for the config value associated with the specified key.
```

Comment:
Why repeat the implemented interface's method's javadoc verbatim?

Also, I'd include an `@Override` on the method.